src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp

```diff
@@ -110,19 +110,53 @@ std::shared_ptr<BrgemmCompiledKernel> BrgemmKernelExecutor::compile_kernel(const
     return compiled_kernel;
 }
 
-void BrgemmKernelExecutor::execute(const BrgemmKernelExecutor* desc, call_args* args) {
-    const auto& kernel = desc->m_kernel;
-    OV_CPU_JIT_EMITTER_ASSERT(kernel, "has nullptr compiler kernel");
-
-    const auto& config = desc->m_config;
-    if (config->is_with_amx) {
-        const auto& amx_tile_config = args->amx_tile_config;
-        if (config->M != amx_tile_config->M || config->K != amx_tile_config->K || config->N != amx_tile_config->N) {
-            amx_tile_config->M = config->M;
-            amx_tile_config->K = config->K;
-            amx_tile_config->N = config->N;
-            cpu::x64::amx_tile_configure(kernel->palette);
+void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr, std::shared_ptr<BrgemmKernelConfig>& config) const {
+    auto get_projected_input_subtensor = [](const snippets::lowered::PortDescriptorPtr& desc) {
+        // Note: for output shape you will need get_preordered_vdims()
+        auto shape = snippets::utils::get_planar_vdims(desc->get_shape(), desc->get_layout());
+        auto subtensor = desc->get_subtensor();
+        OV_CPU_JIT_EMITTER_ASSERT(subtensor.size() <= shape.size() && subtensor.size() == 2,
+                                  "Invalid subtensor + shape combination");
+        auto shape_it = shape.rbegin();
+        for (auto sub_it = subtensor.rbegin(); sub_it != subtensor.rend(); sub_it++, shape_it++) {
+            *sub_it = std::min(*sub_it, *shape_it);
```

Comment:
Sure, let's discuss offline.
The idea here is that subtensors can have dynamic values that indicate that the whole dimension should be processed (usually it's because the blocking is disabled for this axis), so here we replace dynamic values with actual runtime shape dimensions.